### PR TITLE
Traffic Stats base

### PR DESCRIPTION
**Pull request by rurirei**
https://github.com/2dust/v2rayNG/issues/692 https://github.com/2dust/v2rayNG/issues/589 this is the base & core implementation for traffic stats, then two leaving there: UI to show user how much traffic he uses, and your thought of if differ from proxyTraffic and directTraffic to two item.

### CODE REVIEW

File: V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayServiceManager.kt [hunk 1]
```diff
@@ -319,17 +321,19 @@ object V2RayServiceManager {
...
                                 appendSpeedString(text, outboundTags.firstOrNull(), 0.0, 0.0)
                             }
                             appendSpeedString(text, TAG_DIRECT, directUplink / sinceLastQueryInSeconds,
                                     directDownlink / sinceLastQueryInSeconds)
-                            updateNotification(text.toString(), proxyTotal, directDownlink + directUplink)
+                            updateNotification(text.toString(), proxyTotal, directTotal)
                         }
                         lastZeroSpeed = zeroSpeed
                         lastQueryTime = queryTime
+                        AngConfigManager.updateTrafficStats(proxyTotal + directTotal)
```
Comment (yuhan6665) [inline V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayServiceManager.kt:336]: Thanks for your pr. I wanted to add this feature for a while, but was hesitate to do so. For the following reason:
Currently v2rayNG stores the entire list of servers under a single key in the SharedPreference store. This is bad that shared preference is not designed for this. I suspect it is the root cause of #535. Obviously adding frequent update for this list is not a good idea.
I was thinking maybe we should do a data migration, to a storage with one key <-> one server. We should think carefully and consider other optimisation as well.
CC @2dust

Comment (rurirei) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayServiceManager.kt:336]: Preferences behavior needs a rewrite, the main causes is of angConfig & AngConfigManager #663

Comment (yuhan6665) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayServiceManager.kt:336]: We had some chats. We don't plan to add traffic stats at the moment. Though we are looking into possible improvements to server list storage. Including this https://android-developers.googleblog.com/2020/09/prefer-storing-data-with-jetpack.html

Comment (rurirei) [reply V2rayNG/app/src/main/kotlin/com/v2ray/ang/service/V2RayServiceManager.kt:336]: the dataStore does not much than sharedPreferences, recommend to dataBase instead. https://developer.android.com/reference/android/database/package-summary

**Pull request closed without merging**